### PR TITLE
[Maint] Refresh the hub-spoke bicep implementation

### PR DESCRIPTION
**Pull request by ckittel**
* API version updates
* Add missing firewall rules
* Add a few other missing components (diagnostic settings, usage of the route, etc)
* Change how VMs are deployed
  * One VM per spoke.  Always one Linux and one Windows
* Require availability zones
* some master -> main updates
* reduce ARM template parameters
* Style updates to the bicep file to align with other PnP assets
* Added an additional subnet to be a placeholder for private endpoints in the spokes
* Moved from legacy Azure Firewall deployment to Azure firewall policies-based deployments
* Update to modern SKUs and deployment topologies (such as using ephemeral OS, VPN gateway gen 2 w/ AZ support, etc.)
  * Fixes: #194
* Update the image to reflect multiple subnets and also fix the force tunnel to point to the subnet (instead of the vnet)
* Remove the build artifacts/reporting (link to ceapex/pnp is going away)

Verified at there are no required changes to the docs article, though that article is in the refresh queue and there are some improvements to make -- but nothing will break with this commit as it relates to the docs article.

### CODE REVIEW

File: solutions/azure-hub-spoke/README.md [hunk 1]
```diff
@@ -38,54 +38,60 @@ Create a resource group for the deployment.
...
-Run the following command to initiate the deployment with a Linux VM deployed to the first spoke network.
+Run the following command to initiate the deployment with a Linux VM deployed to the first spoke network and a Windows VM deployed to the second spoke network.
 
 ```azurecli-interactive
 az deployment group create \
     --resource-group hub-spoke \
-    --template-uri https://raw.githubusercontent.com/mspnp/samples/master/solutions/azure-hub-spoke/azuredeploy.json \
-    --parameters adminPassword=Password2020! linuxVMCount=1 windowsVMCount=1
+    --template-uri https://raw.githubusercontent.com/mspnp/samples/main/solutions/azure-hub-spoke/azuredeploy.json \
+    --parameters deployVirtualMachines=true adminUsername=azureadmin adminPassword=Password2023!
```
Comment (ckittel) [inline solutions/azure-hub-spoke/README.md:61]: Kill the hardcoded passwords, and let user know it'll prompt for them.

File: solutions/azure-hub-spoke/README.md [hunk 2]
```diff
@@ -16,19 +16,19 @@ description: This sample deploys Azure virtual networks in a hub and spoke confi
...
-| Bicep Template | [![Build Status](https://dev.azure.com/ceapex/pnp/_apis/build/status/deployment-validation/hub-spoke-bastion-bicep?branchName=master)](https://ceapex.visualstudio.com/pnp/_build/latest?definitionId=3729&branchName=master) |
+| ARM Template | [![Build Status](https://dev.azure.com/ceapex/pnp/_apis/build/status/deployment-validation/hub-and-spoke?branchName=main)](https://ceapex.visualstudio.com/pnp/_build/latest?definitionId=3727&branchName=main) |
+| Bicep Template | [![Build Status](https://dev.azure.com/ceapex/pnp/_apis/build/status/deployment-validation/hub-spoke-bastion-bicep?branchName=main)](https://ceapex.visualstudio.com/pnp/_build/latest?definitionId=3729&branchName=main) |
 
 This sample deploys Azure virtual networks in a hub and spoke configuration. An Azure Firewall and Bastion host are also deployed. Optionally, a VPN gateway and sample workload (virtual machines) can be deployed. 
 
 Where applicable, each resource is configured to send diagnostics to an Azure Log Analytics instance.
 
-![Hub and spoke architectural diagram.](images/hub-spoke.png)
+![Hub and spoke architectural diagram.](./images/hub-spoke.png)
```
Comment (ckittel) [inline solutions/azure-hub-spoke/README.md:26]: Consider updating this to match the deployable.

File: solutions/azure-hub-spoke/README.md (hunk 1 above)
Comment (ckittel) [reply solutions/azure-hub-spoke/README.md:59]: I was going to have this always prompt, but that breaks the "no VM" flows (damn shared bicep file).  So, I added a warning.  What do you think @hallihan?

File: solutions/azure-hub-spoke/README.md [hunk 3]
```diff
@@ -16,19 +16,19 @@ description: This sample deploys Azure virtual networks in a hub and spoke confi
 
 | Deployment | Status |
 |---|---|
-| ARM Template | [![Build Status](https://dev.azure.com/ceapex/pnp/_apis/build/status/deployment-validation/hub-and-spoke?branchName=master)](https://ceapex.visualstudio.com/pnp/_build/latest?definitionId=3727&branchName=master) |
-| Bicep Template | [![Build Status](https://dev.azure.com/ceapex/pnp/_apis/build/status/deployment-validation/hub-spoke-bastion-bicep?branchName=master)](https://ceapex.visualstudio.com/pnp/_build/latest?definitionId=3729&branchName=master) |
+| ARM Template | [![Build Status](https://dev.azure.com/ceapex/pnp/_apis/build/status/deployment-validation/hub-and-spoke?branchName=main)](https://ceapex.visualstudio.com/pnp/_build/latest?definitionId=3727&branchName=main) |
```
[the remark this answers is not in the document]
Comment (ckittel) [reply solutions/azure-hub-spoke/README.md]: Removed.

**Review by hallihan — APPROVED** [review]
🚢 Looks good!

**Pull request merged**